addons/mail/static/src/models/composer/composer.js

```diff
@@ -81,6 +82,44 @@ function factory(dependencies) {
             });
         }
 
+        /**
+         * Open the full composer modal.
+         */
+        async openFullComposer() {
+            const attachmentIds = this.attachments.map(attachment => attachment.res_id);
+
+            const context = {
+                default_attachment_ids: attachmentIds,
+                default_body: mailUtils.escapeAndCompactTextContent(this.textInputContent),
+                /* FIXME would need to use suggested_partners here task-2280157 */
+                // default_partner_ids: partnerIds,
+                default_is_log: this.isLog,
+                mail_post_autofollow: true,
+            };
+
+            if (this.thread) {
+                context.default_model = this.thread.model;
+                context.default_res_id = this.thread.id;
+            }
+
+            const action = {
+                type: 'ir.actions.act_window',
+                res_model: 'mail.compose.message',
+                view_mode: 'form',
+                views: [[false, 'form']],
+                target: 'new',
+                context: context,
+            };
+            const options = {
+                on_close: () => {
+                    if (this.constructor.get(this) && this.thread) {
+                        this.thread.loadNewMessages();
+                    }
```

Comment:
Looks almost fine now.

Just one small issue after testing, I noticed that the text of the composer is kept in store even after the message is posted through modal. This is because wizard doesn't go through `postMessage` that does `_reset`.

Also "small" composer content is not sync'ed with the full composer in case of close, it still has the old content.
Ideally both should be sync'ed, but this is out of scope/almost impossible since an action is used.

I think an acceptable alternative for now would be to call `_reset`, either when we open or when we close the action, cc @alexkuhn?

Reply:
Long term maybe the "small composer" should also instantiate a `mail.compose.message` and both should be server sync with it?

Reply:
> I think an acceptable alternative for now would be to call `_reset`, either when we open or when we close the action, cc @alexkuhn?

Yes